src/tracks/tracks.c

```diff
@@ -42,44 +62,60 @@ const Tracks * get_tracks()
 }
 
 
-int add_track(const Track *track, size_t index, int mode)
+enum track_add_result add_track(const Track *track, const size_t index,
+                                const enum track_add_mode mode)
 {
-    int result = TRACK_ADD_RESULT_OK;
+        if (index >= MAX_TRACK_COUNT) {
+                pr_error("Invalid track index\r\n");
+                return TRACK_ADD_RESULT_FAIL;
+        }
+
+        switch (mode) {
+        case TRACK_ADD_MODE_IN_PROGRESS:
+        case TRACK_ADD_MODE_COMPLETE:
+                /* Valid cases.  Carry on */
+                break;
+        default:
+                pr_error_int_msg("Unknown track_add_mode: ", mode);
+                return TRACK_ADD_RESULT_FAIL;
+        }
+
+        static Tracks *g_tracksBuffer;
+        if (g_tracksBuffer == NULL) {
+                terminate_lua();
```

Comment:
with this change writing tracks would shut down and re-start lua.  If there is stateful information in the user's Lua script then it would get reset. This would be a surprise to the user; to them, updating tracks has nothing to do with Lua scripting. IOW, it violates the principal of least surprise.